docs/libcurl/opts/CURLOPT_DEBUGFUNCTION.3

```diff
@@ -79,6 +80,11 @@ The data is protocol data sent to the peer.
 The data is SSL/TLS (binary) data sent to the peer.
 .IP CURLINFO_SSL_DATA_IN
 The data is SSL/TLS (binary) data received from the peer.
+.RE
+
+WARNING: This callback may be called with the curl \fIhandle\fP set to an
+internal handle. If you need to distinguish your handle from internal handles
+then set \fICURLOPT_PRIVATE(3)\fP on your handle. (Added in 8.4.0)
```

Comment:
I think you should move the `(Added)` part to the first sentence, as otherwise it could be read as if the PRIVATE thing is the new addition.

Reply:
Done. I made it two separate paragraphs. Also I added a `DEBUGASSERT(!data->set.private_data);` before the callback is called.

Reply:
If the application only wants to output what is sent/received, it may not care to differentiate which handle is doing it and then it does not need to set `CURLOPT_PRIVATE` at all. So isn't the assert asserting a bit too much?

Reply:
The assert only checks that private data is not set for an internal handle, the code is 
~~~c
      if(data->internal)
        DEBUGASSERT(!data->set.private_data);
~~~

Reply:
Ah, yes that makes sense!